### PR TITLE
feat: [BREAKING_CHANGE] refactor Messenger

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [ ] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

messenger

##### Description of change

- broadcast is changed, it will send to all process
- send is changed, it won't emit message

Closes eggjs/egg#28

### CODE REVIEW

File: test/utils.js [hunk 1]
```diff
@@ -7,6 +7,8 @@ const mm = require('egg-mock');
 const fixtures = path.join(__dirname, 'fixtures');
 const eggPath = path.join(__dirname, '..');
 
+process.env.IS_NODE_DEV_RUNNER = 'true';
```
Comment (popomore) [inline test/utils.js]: @fengmk2 这个能不能改个名字 sendmessage 里

File: test/lib/core/messenger.test.js [hunk 2]
```diff
@@ -98,9 +70,6 @@ describe('test/lib/core/messenger.test.js', () => {
     it('app should accept agent assgin pid message', () => {
       app.expect('stdout', /\[app] agent-to-app agent msg \d+/);
     });
-    it('app should accept itself message', () => {
```
Comment (popomore) [inline test/lib/core/messenger.test.js]: 现在不会发 message 了

File: test/utils.js (hunk 1 above)
Comment (popomore) [inline test/utils.js]: https://github.com/node-modules/sendmessage/pull/2

**Comment by popomore** [comment]
@dead-horse 看看这个改动，broadcast 意义变了，之后修改影响可能会很大。

**Comment by dead-horse** [comment]
CI 挂了？

**Comment by dead-horse** [comment]
+1

**Comment by popomore** [comment]
这个改了之后，mm.app 测 messenger 会有问题，之前想用 sendmessage 中转，发现好多坑。

**Pull request merged**